testdata/p4_14_samples_outputs/09-IPv4OptionsUnparsed-midend.p4

```diff
@@ -77,6 +77,7 @@ parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout
     @name(".parse_ipv4") state parse_ipv4 {
         tmp = packet.lookahead<bit<160>>();
         tmp_hdr_0.setValid();
+        tmp_hdr_0.setValid();
```

Comment:
So it looks like we get redundant setValid calls in some cases?  Harmless, but do we know why this happens?

Reply:
I guess this happens if the user was already setting the header to valid before assigning to it.
We should write an analysis which can remove the useless setValid; it could probably be folded in the def-use analysis.

Reply:
See #2396